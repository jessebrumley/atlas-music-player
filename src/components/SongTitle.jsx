export default function SongTitle() {
  return (
    <div className="text-center">
      <h2 className="text-xl font-bold">Song Title</h2>
      <p className="text-zinc-400">Artist Name</p>
    </div>
  );
}
