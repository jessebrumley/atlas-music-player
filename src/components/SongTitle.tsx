/**
 * SongTitle.tsx
 *
 * Displays the title and artist of the currently playing track.
 * - Uses two different fonts for visual contrast
 * - Expects a `Track` object as the `song` prop
 */
import { Track } from "./Api";

export default function SongTitle({ song }: { song: Track }) {
  return (
    <div>
      <p className="font-fontmain text-2xl text-text">{song.title}</p>
      <p className="font-fontalt text-text">{song.artist}</p>
    </div>
  );
}
