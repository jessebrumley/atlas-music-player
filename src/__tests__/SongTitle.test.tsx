import "@testing-library/jest-dom/vitest";
import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import SongTitle from "../components/SongTitle";
import type { Track } from "../components/Api";

describe("SongTitle component", () => {
  const renderWithTrack = (track: Track) => render(<SongTitle song={track} />);

  it("renders song title and artist with correct styles", () => {
    const mockTrack: Track = {
      id: 1,
      title: "Shimmer",
      artist: "Echo Sun",
      duration: 200,
      audio: "audio.mp3",
      cover: "cover.jpg",
    };

    renderWithTrack(mockTrack);

    expect(screen.getByText("Shimmer")).toBeInTheDocument();
    expect(screen.getByText("Echo Sun")).toBeInTheDocument();

    expect(screen.getByText("Shimmer")).toHaveClass("font-fontmain", "text-2xl", "text-text");
    expect(screen.getByText("Echo Sun")).toHaveClass("font-fontalt", "text-text");
  });

  it("renders alternate track correctly", () => {
    const mockTrack: Track = {
      id: 2,
      title: "Ocean Glow",
      artist: "Neon Drift",
      duration: 180,
      audio: "glow.mp3",
      cover: "glow.jpg",
    };

    renderWithTrack(mockTrack);

    expect(screen.getByText("Ocean Glow")).toBeInTheDocument();
    expect(screen.getByText("Neon Drift")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const mockTrack: Track = {
      id: 3,
      title: "Twilight Breeze",
      artist: "Dream Shapes",
      duration: 240,
      audio: "twilight.mp3",
      cover: "twilight.jpg",
    };

    const { container } = renderWithTrack(mockTrack);
    expect(container).toMatchSnapshot();
  });
});
