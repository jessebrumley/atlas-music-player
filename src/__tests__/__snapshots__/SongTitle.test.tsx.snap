// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle component > matches snapshot 1`] = `
<div>
  <div>
    <p
      class="font-fontmain text-2xl text-text"
    >
      Twilight Breeze
    </p>
    <p
      class="font-fontalt text-text"
    >
      Dream Shapes
    </p>
  </div>
</div>
`;
